scripts/bench_summary.lean

```diff
@@ -0,0 +1,208 @@
+
+The code itself interfaces with the shell to retrieve and process json-data and eventually
+uses `benchOutput`.
+Here is a summary of the steps:
+* retrieve the last comment to the PR (using `gh pr view ...`),
+* check if it was posted by `leanprover-bot`,
+* try to retrieve the source and target commits from the url that the bot posts
+  and store them in `src` and `tgt`,
+* query the speed-center for the benchmarking data (using `curl url`),
+* format and filter various times the returned json-data,
+  saving intermediate steps into `tempFile` (using `jq` multiple times),
+* process the final string to produce a summary (using `benchOutput`),
+* finally post the resulting output (using `gh pr comment ...`).
+-/
+def addBenchSummaryComment (PR : Nat) (repo : String) (tempFile : String := "benchOutput.json") :
+    CommandElabM Unit := do
+  let PR := s!"{PR}"
+  let jq := ".comments | last | select(.author.login==\"leanprover-bot\") | .body"
+  let gh_pr_comments : IO.Process.SpawnArgs :=
+    { cmd := "gh", args := #["pr", "view", PR, "--repo", repo, "--json", "comments", "--jq", jq] }
+  let output ← IO.Process.run gh_pr_comments
+  let frags := output.split (· == '/')
+  let some compIdx := frags.findIdx? (· == "compare") |
+    logInfo "No 'compare' found in URL."
+    return
+  let src := frags.getD (compIdx + 1) ""
+  let tgt := (frags.getD (compIdx + 3) "").takeWhile (· != ')')
+  if (src.length, tgt.length) != (36, 36) then
+    logInfo m!"Found\nsrc: '{src}'\ntgt: '{tgt}'\ninstead of two commit hashes."
+    return
+  dbg_trace s!"Using commits\nsrc: '{src}'\ntgt: '{tgt}'\n"
+  let curlSpeedCenter : IO.Process.SpawnArgs :=
+    { cmd := "curl"
+      args := #[s!"http://speed.lean-fro.org/mathlib4/api/compare/{src}/to/{tgt}?all_values=true"] }
+  let bench ← IO.Process.run curlSpeedCenter
+  IO.FS.writeFile tempFile bench
+  let threshold := s!"{10 ^ 9}"
+  let jq1 : IO.Process.SpawnArgs :=
+    { cmd := "jq"
+      args := #["-r", "--arg", "thr", threshold,
+        ".differences | .[] | ($thr|tonumber) as $th |
+        select(.dimension.metric == \"instructions\" and ((.diff >= $th) or (.diff <= -$th)))",
+        tempFile] }
+  let firstFilter ← IO.Process.run jq1
+  IO.FS.writeFile tempFile firstFilter
+  let jq2 : IO.Process.SpawnArgs :=
+    { cmd := "jq"
+      args := #["-c", "[{file: .dimension.benchmark, diff: .diff, reldiff: .reldiff}]", tempFile] }
+  let secondFilter ← IO.Process.run jq2
+  IO.FS.writeFile tempFile secondFilter
+  let jq3 : IO.Process.SpawnArgs :=
+    { cmd := "jq", args := #["-n", "reduce inputs as $in (null; . + $in)", tempFile] }
+  let thirdFilter ← IO.Process.run jq3
+  let report ← benchOutput thirdFilter
+  IO.println report
+  let add_comment : IO.Process.SpawnArgs :=
+    { cmd := "gh", args := #["pr", "comment", PR, "--repo", repo, "--body", report] }
+  let _ ← IO.Process.run add_comment
+
+--run_cmd addBenchSummaryComment putPR "leanprover-community/mathlib4"
```

Comment:
I'm assuming this can be deleted?
```suggestion
```

Reply:
Actually, line28 of the workflow 

```
          sed "s=--\(run_cmd .*\) putPR=\1 ${PR}=" scripts/bench_summary.lean |
```

uncomments this line and makes it live in CI.

Reply:
Oh wow! It would be good to have a normal comment above this line explaining this.

Reply:
I modified the workflow, but both, adding a comment in the file, but also changing the `sed` line to a "`cat` + printf` pair that *add* the extra line, rather than uncommenting the one present in the file.